pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/SimpleConsumerWrapper.java

```diff
@@ -85,8 +86,8 @@
    * A Kafka protocol error that indicates a situation that is not likely to clear up by retrying the request (for
    * example, no such topic or offset out of range).
    */
-  public static class PermanentConsumerException extends RuntimeException {
-    public PermanentConsumerException(Errors error) {
+  public static class KafkaPermanentConsumerException extends StreamConsumerExceptions.PermanentConsumerException {
```

Comment:
Can we get rid of this class altogether and just use StreamConsumerExceptions.PermanentConsumerException?

Reply:
Refactored it to have PermanentConsumerException and TransientConsumerException in their own classes. Exceptions thrown from the consumer would be generic, but wrapped around a specific exception defined by the stream if required.